Sources/StreamChat/Controllers/ListDatabaseObserver.swift

```diff
@@ -168,8 +170,37 @@ class ListDatabaseObserver<Item, DTO: NSManagedObject> {
                 // callback which can cause a race condition when the object is already being deinited on a different thread.
                 guard let self = self else { return }
 
-                self._items.reset()
-                self.onChange?($0)
```

Comment:
Very often than not, calling these two lines in the main thread results `performAndWait()` to be called in the main thread. This PR is trying to avoid that in this particular control path.